.github/actions/setup-caddy/action.yml

```diff
@@ -0,0 +1,13 @@
+name: Setup Caddy server
+runs:
+  using: composite
+  steps:
+    - shell: bash
+      run: |
+        sudo apt-get install -y debian-keyring debian-archive-keyring apt-transport-https
+        curl -1sLf 'https://dl.cloudsmith.io/public/caddy/stable/gpg.key' | sudo gpg --dearmor -o /usr/share/keyrings/caddy-stable-archive-keyring.gpg
```

Comment:
hmmm not sure about this. cc @iluuu1994

Reply:
We need to install Caddy locally (not in a container). Caddy creates a TLS certificate and update the local trust store so we don't have to deal with self-signed certificates (HTTPS is need for HTTP/2).

We could also just download the binary instead of using the `.deb` but then we will have to deal with updates manually (although it's probably not a big deal to use an outdated version in a CI, thought).

Reply:
I m not critiquing the solution it brings, more the fact to rely on another provider which can bring its own set of issues (reliability, changes of policy, ...). It s just a question ; but iluuu1994 will be a better judge on this.

Reply:
Sure, sorry if my comment was misleading, it was to explain the choice and propose an alternative (download the binary directly).

Reply:
ok makes more sense.

Reply:
Looks fine. Stability is not a big concern, if it breaks we can fix it or remove it.